deepsensor/data/loader.py

```diff
@@ -1279,24 +1487,113 @@ def __call__(
                 Task object or list of task objects for each date containing
                 the context and target data.
         """
-        if isinstance(date, (list, tuple, pd.core.indexes.datetimes.DatetimeIndex)):
-            return [
-                self.task_generation(
-                    d,
-                    context_sampling,
-                    target_sampling,
-                    split_frac,
-                    datewise_deterministic,
-                    seed_override,
+        assert patch_strategy in [None, "random", "sliding"], (
+            f"Invalid patch strategy {patch_strategy}. "
+            f"Must be one of [None, 'random', 'sliding']."
+        )
+
+        if patch_strategy is None:
```

Comment:
I moved the logic to the `_call_` function, however, there is quiet a bit of code redundancy  because:
- checking separate sampling strategies
- checking whether one supplies a single data or a sequence of date that determines whether a `Task` or a `list[Task]` is returned

So that can be made more concise